### ISSUE TITLE
Incorrect links when using HTML base tag

### ISSUE THREAD

**Issue by tomyam1** [tscanlin/tocbot#154]
Since

> Anchor tags within a page, e.g. <a href="#anchor">anchor</a>, are resolved by using the base URL as the reference and trigger an HTTP request to the base URL.

Source: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base#Anchor_elements

The links [generated by toc](https://github.com/tscanlin/tocbot/blob/3aec534cf7546194e75a891263cc18d1fc6b4567/src/js/build-html.js#L90) link to that anchor on the base URL.

Tried to fix locally by making the href absolute:

```ts
$('.toc-link').each(function (this: HTMLAnchorElement) {
        const a = document.createElement('a');
        a.href = document.location.href;
        a.hash = this.hash;
        this.href = a.href;
    });
```

but it failed because tocbot relies on the href, e.g. [in here](https://github.com/tscanlin/tocbot/blob/3aec534cf7546194e75a891263cc18d1fc6b4567/src/js/build-html.js#L207)

The best result I got with
```ts
Tocbot.init({
       // ...
        onClick: function (this: HTMLAnchorElement) {
            const a = document.createElement('a');
            a.href = document.location.href;
            a.hash = this.hash;
            document.location.href = a.href;

            Tocbot.refresh();

            return false;
        } as any, //as any because it wrongly only allows boolean
    });
```

**Comment by tscanlin** [comment]
why can't the links just have the hash instead of the full URL?

**Comment by tomyam1** [comment]
Because

> Anchor tags within a page, e.g. anchor, are resolved by using the base URL as the reference and trigger an HTTP request to the base URL.

That means that if you have a `<base href="/foor/"` then a link to `#anchor` would resolve to `/foor/#anchor` which probably is in another.

**Comment by tscanlin** [comment]
but does that take you to a different page or the same page? because if its on the same page you don't need a base url, and if its on a different page then the browser handles scrolling users to the proper part of the page.

**Comment by tscanlin** [comment]
closing due to no response.

**Comment by devonmitton** [comment]
I know this is closed, but I wanted to add some details since I'm seeing the same issue.

For more context, I'm working within a platform that I don't have full control of, and it uses a `<base href="..." />` tag across the entire site to ensure all relative links point to the right location for it's own assets (images, css, js, etc.).

The side-effect of `base` tags is that in-page anchors without an absolute path will use the url in the `base` tag.

**Scenario**
```
 Website Base URL: https://mywebsite.com/path
Page using Tocbot: https://mywebsite.com/path/to/page
 Link from Tocbot: <a href="#anchor-from-tocbot">...</a>
Expected behavior: https://mywebsite.com/path/to/page#anchor-from-tocbot
  Result b/c base: https://mywebsite.com/path#anchor-from-tocbot
```

Is there a way to change the links created by Tocbot to use an absolute path? E.g.:
```html
<!-- Link generated by Tocbot -->
<a href="https://mywebsite.com/path/to/page#anchor-from-tocbot">...</a>
```

Thanks!

**Comment by tscanlin** [comment]
Hey, thanks for the context on this and explaining the expectation vs reality so well. That makes a lot of sense and I'm happy to help.

I meant to open a PR but pushed on accident. Anyway, its a pretty minor change and I tested it locally. You should see the fix in `tocbot@4.11.0`. Thanks for finding and responding to this original issue, please ping me if you have any other problems.

Hope this helps you!
https://github.com/tscanlin/tocbot/commit/dffdc28b68bb948d2ad767882f255003ea020a08

**Comment by devonmitton** [comment]
Works great! Thanks so much 👍
